single-line-diagram-core/src/main/java/com/powsybl/sld/layout/PositionVoltageLevelLayout.java

```diff
@@ -88,17 +92,31 @@ private void adaptPaddingToSnakeLines(LayoutParameters layoutParam) {
     }
 
     private void calculateBusNodeCoord(VoltageLevelGraph graph, LayoutParameters layoutParam) {
-        graph.getNodeBuses().forEach(nb -> nb.calculateCoord(layoutParam, graph.getFirstBusY()));
+        graph.getNodeBuses().forEach(nb -> calculateNodeCoord(nb, layoutParam, graph.getFirstBusY()));
+    }
+
+    private void calculateNodeCoord(BusNode busNode, LayoutParameters layoutParameters, double firstBusY) {
+        double elementaryWidth = layoutParameters.getCellWidth() / 2;
+        double busPadding = layoutParameters.getBusPadding();
+        Position position = busNode.getPosition();
+        busNode.setCoordinates(position.get(H) * elementaryWidth + busPadding,
+            firstBusY + position.get(V) * layoutParameters.getVerticalSpaceBus());
+        busNode.setPxWidth(position.getSpan(H) * elementaryWidth - 2 * busPadding);
     }
 
     private void calculateCellCoord(VoltageLevelGraph graph, LayoutParameters layoutParam) {
-        graph.getBusCellStream().forEach(cell -> cell.calculateCoord(layoutParam, createLayoutContext(graph, cell, layoutParam)));
-        graph.getShuntCellStream().forEach(cell -> cell.calculateCoord(layoutParam, null));
+        CalculateCellCoordVisitor calculateCellCoordVisitor = new CalculateCellCoordVisitor(layoutParam);
+        graph.getBusCellStream().forEach(cell -> {
+            calculateCellCoordVisitor.setLayoutContext(createLayoutContext(graph, cell, layoutParam));
+            cell.accept(calculateCellCoordVisitor);
+        });
+        calculateCellCoordVisitor.setLayoutContext(null);
```

Comment:
You'd rather use a new `CalculateCellCoordVisitor` for each bus cell and add a constructor with LayoutContext argument

Reply:
ok